src/System.Net.Primitives/tests/FunctionalTests/CredentialCacheTest.cs

```diff
@@ -392,24 +432,99 @@ public static void DefaultCredentials_Get_Success()
             Assert.Equal(String.Empty, c.Password);
             Assert.Equal(String.Empty, c.Domain);
         }
-
-        [Fact]
-        [SkipOnTargetFramework(TargetFrameworkMonikers.Netcoreapp | TargetFrameworkMonikers.Uap)]
-        public static void AddRemove_UriAuthenticationTypeDefaultCredentials_Success_net46()
+        
+        [Theory]
+        [MemberData(nameof(StandardAuthTypeWithNetworkCredential))]
+        [MemberData(nameof(CustomAuthTypeWithCustomNetworkCredential))]
+        public static void Add_UriAuthenticationType_Success(string authType, NetworkCredential nc)
         {
-            NetworkCredential nc = CredentialCache.DefaultNetworkCredentials as NetworkCredential;
+            // Default credentials cannot be supplied for the Basic authentication scheme.
+            if (string.Equals(authType, authenticationTypeBasic, StringComparison.OrdinalIgnoreCase) && (nc == CredentialCache.DefaultNetworkCredentials))
+            {
+                return;
+            }
+            
             CredentialCache cc = new CredentialCache();
-
-            // The full framework throw System.ArgumentException with the message
-            // 'Default credentials cannot be supplied for the authenticationType1 authentication scheme.'
-            AssertExtensions.Throws<ArgumentException>("authType", () => cc.Add(uriPrefix1, authenticationType1, nc));
+            
+            // .NET Framework and .NET Core have different behaviors for Digest when default NetworkCredential is used.
+            if (string.Equals(authType, authenticationTypeDigest, StringComparison.OrdinalIgnoreCase) && (nc == CredentialCache.DefaultNetworkCredentials))
+            {
+                if (PlatformDetection.IsFullFramework)
+                {
+                    // In .NET Framework, when authType == Digest, if WDigestAvailable == true, it will pass the validation.
+                    // if WDigestAvailable == false, it will throw ArgumentException.
+                    // It is not possible to easily determine if Digest is supported or not on .NET Framework. So, we will skip the test.
+                    return;
+                }
+                else
+                {
+                    // In .NET Core, WDigestAvailable will always be false (we don't support it).
+                    // It will always throw ArgumentException.
+                    AssertExtensions.Throws<ArgumentException>("authType", () => cc.Add(uriPrefix1, authType, nc));
+                    return;
+                }
```

Comment:
After looking at your code now, you need to remove the 'return' statements.  You don't want the test to return at that point since you have more Assert.* statements below